recipes/sagemaker-code-editor/meta.yaml

```diff
@@ -0,0 +1,68 @@
+{% set name = "sagemaker-code-editor" %}
+{% set version = "0.1.0" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version|replace("-", "_") }}
+
+source:
+  # github repo set-up is in progress, once ready we will use the github release url here instead of the s3 url
+  url: https://sagemaker-code-editor-1696292125.s3.us-west-2.amazonaws.com/code-editor.tar.gz
+  sha256: 9effda827c45e1fd8dfc534ce310791ee46070d818ede89abf19e3ec9c8906d0
+  folder: sagemaker-code-editor
+
+build:
+  number: 0
+  skip: true  # [not linux]
+
+requirements:
+  build:
+    # dependencies needed to build open source microsoft/vscode - https://github.com/microsoft/vscode/wiki/How-to-Contribute 
```

Comment:
This might be a silly question, but why are we rebuilding vscode in this recipe, rather than having a proper vscode build that can be reused as a component?

I see that we have a feedstock for [openvscode-server](https://github.com/conda-forge/openvscode-server-feedstock), which looks very related, and has very similar build requirements (see [recipe](https://github.com/conda-forge/openvscode-server-feedstock/blob/main/recipe/meta.yaml))

CC @conda-forge/openvscode-server

Reply:
@h-vetinari , For Amazon SageMaker's use case, we have some changes on top of the open source [Code - OSS](https://github.com/microsoft/vscode). These customizations are not available on openvscode-server and can't be added there since the use cases are very specific to our product.

Here's the overall plan to clarify:
1. We're setting up a public github repo `sagemaker-code-editor` which will pull in microsoft/vscode as a submodule and we will add our changes as patches. This set up is in progress and will be up shortly.
2. The patched source code is hosted on that repo using Github releases (in current PR, this is hosted on s3)
3. This conda-forge recipe uses that Github release as the source and builds it (it was recommended in [earlier comments](https://github.com/conda-forge/staged-recipes/pull/24163#issuecomment-1771219792) in this PR that we build from source)

Reply:
`conda-build` supports having patches on top of the source. You can see how we have a similar setup in [conda-standalone](https://github.com/conda-forge/conda-standalone-feedstock/blob/main/recipe/meta.yaml): your first source is the repo with patches, the second one would be the upstream source to be patched. You make the second source tag or sha match your submodule. This way, everything is as transparent as possible in the recipe.

Reply:
@jaimergp - yes, I did see that conda-build supports patches

Re-iterating options below to be on same page:
Option 1 (suggestion above) - we would still set up our repo with patches but apply the patches in this conda-forge recipe
Option 2 (current plan) - applying the patches in github and consuming the patched source code from the recipe

I'm still inclined towards Option 2 since: 
- when we're looking to automate the conda-forge package updates, it'll be easier if there's just once source that follows semver versioning. For different commits of the submodule, we can make sure patches are compatible in the github repo itself - and the recipe can just consume it
- we would need to apply patches in the github repo anyway since we need to run tests to make sure that patches are not broken

From a transparency point of view, Option 2 is just one step extra - users navigating to the github repo linked in this recipe. So, that seems manageable. Let me know if this makes sense.

Reply:
@jaimergp Can you please let us know if you think the repo set up blocks the merge of this PR ? 

We're actively working on the github repo set up but it's taking time due to company internal processes. 
We would prefer to make subsequent improvements once the feedstock is created since there's some work blocked on this package being available on conda-forge.

Reply:
It's not a blocker but it's definitely unconventional. IMO we could go ahead with the condition of eventually providing the repository. This should go in an issue in the feedstock.

Reply:
@jaimergp - thank you. Yes, I'll create a github issue once the feedstock is created.